src/Engine/IEngine.cs

```diff
@@ -2,7 +2,7 @@
 
 namespace CleanLiving.Engine
 {
-    public interface IEngine<TTime>
+    public interface IEngine<TTime> : IDisposable
```

Comment:
I don't think the `IEngine` *needs* to be IDisposable. The implementation certainly does because it holds onto disposables... up for discussion but I don't consider it necessary here.